src/App.vue

```diff
@@ -731,18 +751,29 @@ export default {
       }
     },
 
-    storeDirectors (response: any): void {
-      const directors = response?.data?.directors
-      if (directors) {
-        const directorsSorted = directors.sort(this.fieldSorter(['lastName', 'firstName', 'middleName']))
-        for (var i = 0; i < directorsSorted.length; i++) {
-          directorsSorted[i].id = i + 1
-          directorsSorted[i].isNew = false
-          directorsSorted[i].isDirectorActive = true
+    storeParties (response: any): void {
+      const parties = response?.data?.parties
+      if (parties) {
+        const directorsList = parties?.filter(partiesArr => partiesArr.roles?.includes(Roles.DIRECTOR))
+        const custodianList = parties?.filter(partiesArr => partiesArr.roles?.includes(Roles.CUSTODIAN))
+
+        // Directors
+        if (directorsList) {
+          const directorsListSorted = directorsList.sort(this.fieldSorter(['lastName', 'firstName', 'middleName']))
+          for (var i = 0; i < directorsListSorted.length; i++) {
+            directorsListSorted[i].id = i + 1
+            directorsListSorted[i].isNew = false
+            directorsListSorted[i].isDirectorActive = true
+          }
+          this.setDirectors(directorsListSorted)
+        }
+
+        // Custodians
+        if (custodianList) {
+          this.setCustodians(custodianList)
```

Comment:
Instead of 3 different arrays, would it make sense to store just the parties and then provide getters to extract just the directors or custodians?

Reply:
Possibly
The directors are structured a bit different, with some additional properties assigned locally as you can see above. Which is why we also preserved that existing directors interface as well. 
They could be stored in the same place but the functionality to isolate the types between each other has to be done somewhere i suppose, so it makes sense to do it up front.

Reply:
OK.